appservice/src/TunnelProxy.ts

```diff
@@ -147,39 +172,48 @@ export class TunnelProxy {
     }
 
     private async checkTunnelStatus(): Promise<void> {
-        return new Promise<void>((resolve: () => void, reject: () => void): void => {
-            const statusOptions: request.Options = {
-                uri: `https://${this._client.kuduHostName}/AppServiceTunnel/Tunnel.ashx?GetStatus`,
-                headers: {
-                    'User-Agent': 'vscode-azuretools'
-                },
-                auth: {
-                    user: this._publishCredential.publishingUserName,
-                    pass: this._publishCredential.publishingPassword
-                }
-            };
-
-            const statusCallback: request.RequestCallback = (error: string, response: request.Response, body: string): void => {
-                if (error) {
-                    ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, error: ${error}`);
-                    reject();
-                } else if (response.statusCode === 200) {
-                    ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, body: ${body}`);
-
-                    // SUCCESS:2222 means that the default ssh tunnel has not been replaced yet
-                    if (isString(body) && body.startsWith('SUCCESS') && body !== 'SUCCESS:2222') {
-                        resolve();
-                    } else {
-                        reject();
-                    }
-                } else {
-                    ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, unexpected response: ${response.statusCode} - ${response.statusMessage}`);
-                    reject();
-                }
-            };
+        // return new Promise<void>((resolve: () => void, reject: (failure: ICheckTunnelStatusFailure) => void): void => {
+        const statusOptions: requestP.Options = {
+            uri: `https://${this._client.kuduHostName}/AppServiceTunnel/Tunnel.ashx?GetStatus&GetStatusAPIVer=2`,
+            headers: {
+                'User-Agent': 'vscode-azuretools'
+            },
+            auth: {
+                user: this._publishCredential.publishingUserName,
+                pass: this._publishCredential.publishingPassword
+            }
+        };
 
-            request.get(statusOptions, statusCallback);
-        });
+        let tunnelStatus: ITunnelStatus;
+        try {
+            // tslint:disable-next-line:no-unsafe-any
+            const responseBody: string = await requestP.get(statusOptions);
+            ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, body: ${responseBody}`);
+
+            // tslint:disable-next-line:no-unsafe-any
+            tunnelStatus = JSON.parse(responseBody);
+        } catch (error) {
+            const parsedError: IParsedError = parseError(error);
+            ext.outputChannel.appendLine(`[WebApp Tunnel] Checking status, error: ${parsedError.message}`);
+            throw new Error(`Error getting tunnel status: ${parsedError.errorType}`);
```

Comment:
Did you mean to use `errorType` here or should this be `message`?

Reply:
This is what I intended. I tried out passing this method invalid JSON and got a convoluted error that would eventually show to the user as:

Unable to establish connection to application: Error getting tunnel status: Unexpected token d in JSON at position 1

Using errorType gives something a little nicer:

Unable to establish connection to application: Error getting tunnel status: SyntaxError

And then the full error message can be found in the output window.